### PR TITLE
Hide delete new

### PR DESCRIPTION
**Pull request by kaykakaraka**
cannot see delete button for posts that are not yours
can see delete button for posts that are yours

### CODE REVIEW

File: cypress/integration/user_cannot_delete_other_users_post_spec.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+describe('delete button',() => {
```
Comment (michaelcychan) [inline cypress/integration/user_cannot_delete_other_users_post_spec.js:1]: Good use of beforeEach, tests are clear and easy to understand. Nice!

File: controllers/posts.js [hunk 2]
```diff
@@ -7,6 +7,11 @@ const PostsController = {
         throw err;
       }
       posts.reverse(); // reorders posts, so newest post is always at the top of the list
+      posts.forEach((post) => {
+        if (post.user === req.session.user.email) {
+          Object.assign(post, {canDelete: true})
```
Comment (michaelcychan) [inline controllers/posts.js:12]: a clever way to assign the canDelete value.

**Review by michaelcychan — APPROVED** [review]
Clever way to add and use the canDelete indicator.
Amazing

File: cypress/integration/user_cannot_delete_other_users_post_spec.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  beforeEach(() => {
+    cy.task('emptyUsers').then(() => {
```
Comment (Hafizalimi7) [inline cypress/integration/user_cannot_delete_other_users_post_spec.js:3]: good amount of test

**Review by Hafizalimi7 — APPROVED** [review]

**Pull request merged**